src/main/java/com/casper/sdk/model/deploy/DelegatorKind.java

```diff
@@ -1,15 +1,36 @@
 package com.casper.sdk.model.deploy;
 
+import com.casper.sdk.model.key.PublicKey;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.annotation.JsonSubTypes;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.math.BigInteger;
 
 /**
  * Auction bid variants. Kinds of delegation bids
  *
  * @author carl@stormeye.co.uk
  */
+@Getter
+@Setter
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.WRAPPER_OBJECT)
 @JsonSubTypes({
         @JsonSubTypes.Type(value = DelegatorKindPurse.class, name = "Purse"),
         @JsonSubTypes.Type(value = DelegatorKindPublicKey.class, name = "PublicKey")})
-public interface DelegatorKind {}
+public class DelegatorKind extends SeigniorageAllocation {
```

Comment:
Why are you adding these to DelegatorKind it does not implement these in the Rust code